javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/declarations/JavaParserInterfaceDeclaration.java

```diff
@@ -310,11 +310,12 @@ public AccessSpecifier accessSpecifier() {
 
     private ResolvedReferenceType toReferenceType(ClassOrInterfaceType classOrInterfaceType) {
         SymbolReference<? extends ResolvedTypeDeclaration> ref = null;
-        if (classOrInterfaceType.toString().indexOf('.') > -1) {
-            ref = typeSolver.tryToSolveType(classOrInterfaceType.toString());
+        String typeName = classOrInterfaceType.getNameAsString();
+        if (typeName.indexOf('.') > -1) {
```

Comment:
we should probably have some utility class for this sort of things

Reply:
Well, what does it do?

Reply:
I mean, to split a qualified name in its component or identify if a name is qualified

Reply:
Oh, that actually sounds like `JavaParser.parseName(String)`